src/types.tsx

```diff
@@ -0,0 +1,17 @@
+export interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+export interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+export interface TodoWithUser extends Todo {
+  user: User | null
+}
```

Comment:
it is better to create separate Types folder with separate type files for User and Todo